core/src/consensus.rs

```diff
@@ -931,7 +928,12 @@ impl Tower {
                 if !voting_from_genesis {
                     // Unless we're voting since genesis, slots_in_tower must always be older than last checked_slot
                     // including all vote slot and the root slot.
-                    assert!(*slot_in_tower < checked_slot)
+                    assert!(
```

Comment:
I'm adding tests before adjusting this assert!.

Reply:
Hi! I'm catching this assertion when I'm running a new validator: https://github.com/solana-labs/solana/issues/13128